src/stores/ScannerStore.js

```diff
@@ -81,6 +82,24 @@ export default class ScannerStore extends Container<ScannerState> {
   async setParsedData(strippedData, accountsStore) {
     const parsedData = await constructDataFromBytes(strippedData);
 
+    const chainName = checkAddress(parsedData.data.account, 2)[0] ? 'Kusama' : checkAddress(parsedData.data.account, 42)[0] ? 'Polkadot' : null;
```

Comment:
I'm worried that this logic and what follows isn't generic and needs to be extended as soon as there's a new network.
Can't we use `NETWORK_LIST[genesisHash].prefix` ?

Reply:
I agree it's not ideal, but updates will be needed when there's a new network regardless.

By the way this is for the payloads without a genesisHash such as signing a message or a hash.